map/framework.cpp

```diff
@@ -169,6 +170,38 @@ void CancelQuery(weak_ptr<search::ProcessorHandle> & handle)
     queryHandle->Cancel();
   handle.reset();
 }
+
+string GetStreet(search::ReverseGeocoder const & coder, Index const & index, FeatureType const & ft)
+{
+  auto const & editor = osm::Editor::Instance();
+  string streetName;
+
+  if (editor.GetFeatureStatus(ft.GetID()) == osm::Editor::FeatureStatus::Created)
+  {
+    VERIFY(editor.GetEditedFeatureStreet(ft.GetID(), streetName), ("Feature is in editor."));
+    return streetName;
+  }
+  search::ReverseGeocoder::Address address;
+  coder.GetNearbyAddress(ft.GetCenter(), address);
+  return address.GetStreetName();
+}
+
+string MakeSearchBookingUrl(Index const & index, booking::Api const & bookingApi,
+                            FeatureType const & ft)
+{
+  search::ReverseGeocoder const coder(index);
+  string name;
+  ft.GetReadableName(name);
+
+  my::Cancellable unused;
+  search::VillagesCache unusedCache(unused);
+  search::LocalityFinder finder(index, unusedCache);
```

Comment:
I strongly suggest you to wrap this in the class and put in the framework, as LocalityFinder has a cache and it's not a good idea to drop the contents of the cache at the exit from the function.